src/Mvc/Mvc.Analyzers/src/Startup/StartupAnalyzer.Diagnostics.cs

```diff
@@ -16,5 +16,15 @@ public partial class StartupAnalzyer : DiagnosticAnalyzer
             DiagnosticSeverity.Warning, 
             isEnabledByDefault: true,
             helpLinkUri: "https://aka.ms/YJggeFn");
+
+        internal readonly static DiagnosticDescriptor BuildServiceProviderShouldNotCalledInConfigureServicesMethod = new DiagnosticDescriptor(
+            "MVC1007",
+            "Cannot use BuildServiceProvider in ConfigureServices.",
+            "BuildServiceProvider called by runtime,don't call this method in your application code directly",
+            "Design",
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            helpLinkUri: "https://github.com/aspnet/AspNetCore/issues/11727"
```

Comment:
@davidfowl suggestions on where this should point to?

Reply:
I would assume these should all point to a single docs article via an aka.ms link

Reply:
This should be an aka.ms and it should point to a docs article.

Reply:
@wu-yafeng - we'll get back to you with a URL to put here

Reply:
@wu-yafeng - please make this point to `https://aka.ms/AA5k895`. I'll write a docs article as part of this release.